Api/Api Martijn/app.js

```diff
@@ -1,38 +1,226 @@
+    var measureParts = [2,4,6,7,12,13]
+
+    return await jsCalculateLineLength(yLine, segmentation, measureParts);
+}
+
+async function jsGetHipYLine(image){
+    var poseData = await jsConvertImageElementToPoseData(image);
+
+    return Math.round((poseData.leftHip.position.y + poseData.rightHip.position.y)/2);
+}
+
+//#endregion
+
+//#region Waist
+
+async function jsMeasureWaist(imageInformation){
+    //const waistYLine = jsGetWaistYLine(imageInformation.frontImage);
+    //const waistFrontSize = jsMeasureWaistFront(waistYLine, imageInformation.frontImageSegmentation);
+    //const waistSideSize = jsMeasureWaistSide(waistYLine, imageInformation.sideImageSegmentation);
+
+    //return jsCalculatePerimeter(waistFrontSize, waistSideSize);
+    return 0;
+}
+
+async function jsMeasureWaistFront(yLine, segmentation){
+
+    var measureParts = []
+
+    return await jsCalculateLineLength(yLine, segmentation, measureParts);
+}
+
+async function jsMeasureWaistSide(yLine, segmentation){
+
+    var measureParts = []
+
+    return await jsCalculateLineLength(yLine, segmentation, measureParts);
+}
+
+async function jsGetWaistYLine(image){
+
+}
+
+//#endregion
+
+async function jsCalculateLineLength(yLine, segmentation, measureParts){
+    var startPoint = segmentation.width * yLine;
+    var endPoint = startPoint + segmentation.width;
+    var outlinePixelNumber = [];
+
+    // Here the code cycles througt the point and detects a transition between parts
+    for(var i = startPoint; i < endPoint; i++){
+        var currentPixel = segmentation.data[i];
+        var nextPixel = segmentation.data[i+1];
+
+        //It checks if there is a transition between measureParts a other parts
+        if((measureParts.includes(currentPixel) && !measureParts.includes(nextPixel)) || (measureParts.includes(nextPixel) && !measureParts.includes(currentPixel))){
+            //It adds the point to the collection of points
+            outlinePixelNumber.push(i);
+        }
+    }
+
+    outlinePixelNumber.sort((a,b) => b - a);
+
+    return outlinePixelNumber[0] - outlinePixelNumber[outlinePixelNumber.length-1];
+}
+
+async function jsCalculatePerimeter(width, depth){
+    var a = width / 2;
+    var b = depth / 2;
+
+    //p≈ π(3(a+b)- √((3a+b)(a+3b)))
+    return Math.PI * (3 * (a + b) - Math.sqrt((3 * a + b) * (a + 3 * b)))
+}
+
+async function jsConvertImageElementToPoseData(image){
+    var segmentation = await jsGetSegmentation(image);
+    var poses = segmentation.allPoses;
+    var pose = poses[0];
+
+    var poseData = new BodyParts();
+    pose.keypoints.forEach((keyPoint) => {
+        poseData[keyPoint.part] = keyPoint;
+    });
+
+    return poseData;
+}
+
+class BodyParts{
+    nose 
+    leftEye 
+    rightEye 
+    leftEar 
+    rightEar 
+    leftShoulder 
+    rightShoulder 
+    leftElbow 
+    rightElbow 
+    leftWrist 
+    rightWrist 
+    leftHip 
+    rightHip 
+    leftKnee 
+    rightKnee 
+    leftAnkle 
+    rightAnkle
+}
```

Comment:
Error

> api@1.0.0 start \Project-D\Api\Api Martijn
> node app.js

\Project-D\Api\Api Martijn\app.js:210
    leftEye
    ^^^^^^^

SyntaxError: Unexpected identifier
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! api@1.0.0 start: `node app.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the api@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Gebruiker\AppData\Roaming\npm-cache\_logs\2020-05-17T17_15_04_705Z-debug.log
The terminal process terminated with exit code: 1